packages/salesforcedx-vscode-core/test/vscode-integration/commands/forceAuthDevHub.test.ts

```diff
@@ -27,6 +32,62 @@ describe('Force Auth Web Login for Dev Hub', () => {
   });
 });
 
+describe('configureDefaultDevHubLocation on processExit of ForceAuthDevHubExecutor', () => {
+
+  let getDefaultDevHubUsernameStub: SinonStub;
+  let setGlobalDefaultDevHubStub: SinonStub;
+  const authWebLogin = ForceAuthDevHubExecutor.prototype
+
+  beforeEach(() => {
+    getDefaultDevHubUsernameStub = stub(OrgAuthInfo, 'getDefaultDevHubUsernameOrAlias');
+    setGlobalDefaultDevHubStub = stub(authWebLogin, 'setGlobalDefaultDevHub');
+  });
+
+  afterEach(() => {
+    getDefaultDevHubUsernameStub.restore();
+    setGlobalDefaultDevHubStub.restore();
+
+  });
+
+  it('Should set global dev hub if there is no global already, but a local has been defined', async () => {
+
+    getDefaultDevHubUsernameStub.onCall(0).returns(undefined);
+    getDefaultDevHubUsernameStub.onCall(1).returns("test@test.com");
+
+    await authWebLogin.configureDefaultDevHubLocation();
+
+    expect(setGlobalDefaultDevHubStub.called).to.equal(true);
+    expect(getDefaultDevHubUsernameStub.calledWith(false, ConfigSource.Global)).to.equal(true);
+    expect(getDefaultDevHubUsernameStub.calledWith(false, ConfigSource.Local)).to.equal(true);
+    expect(getDefaultDevHubUsernameStub.calledTwice).to.be.true;
+  });
+
+  it('Should do nothing if there is no local dev hub to refer to', async () => {
+
+    getDefaultDevHubUsernameStub.returns(undefined);
+
+    await authWebLogin.configureDefaultDevHubLocation();
+    expect(setGlobalDefaultDevHubStub.called).to.equal(false);
+    expect(getDefaultDevHubUsernameStub.calledTwice).to.equal(true);
+    expect(getDefaultDevHubUsernameStub.calledWith(false, ConfigSource.Global)).to.equal(true);
+    expect(getDefaultDevHubUsernameStub.calledWith(false, ConfigSource.Local)).to.equal(true);
+  });
+
+  it('Should call set and write on the config file', async () => {
+
+    const configWrite = stub(ConfigFile.prototype, 'write');
+    const configSet = stub(ConfigFile.prototype, 'set');
+    setGlobalDefaultDevHubStub.restore();
```

Comment:
Probably don't need this line since it's covered in the `afterEach` already

Reply:
actually still needed because while the afterEach does indeed restore it, after 'afterEach' is called, 'beforeEach' is called again which instantiates it

Reply:
Ah, so is this because you don't want the `setGlobalDefaultDevHub` method to be stubbed for this test case?

Reply:
exactly

because if it's stubbed then none of the methods inside it get called, so that stub was getting in the way of calls to the ConfigFile method calls